File: backend-python/com/gxs/bank/repository/FixedDepositRepository.py
from com.gxs.bank.model.FixedDeposit import FixedDeposit
from com.gxs.bank.repository._base import BaseRepository


class FixedDepositRepository(BaseRepository):
    def __init__(self, db):
        super().__init__(db, FixedDeposit)

    def findByUserId(self, userId: str):
        return self.db.query(FixedDeposit).filter(FixedDeposit.userId == userId).all()

File: backend/src/main/java/com/gxs/bank/repository/FixedDepositRepository.java
package com.gxs.bank.repository;

import com.gxs.bank.model.FixedDeposit;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface FixedDepositRepository extends JpaRepository<FixedDeposit, UUID> {
    List<FixedDeposit> findByUserId(UUID userId);
}
